### PR TITLE
Mlops-97 Encryption for jobs

### PR DESCRIPTION
**Pull request by JacekHebda**

### CODE REVIEW

**Review by michalmrazek — APPROVED** [review]
Looking good to me

File: src/wanna/core/services/jobs.py [hunk 1]
```diff
@@ -430,7 +433,7 @@ def write_manifest(
         Returns:
             Path: Path where resource manifest was saved to
         """
-
+        encryption_spec_key_name = self.config.gcp_profile.kms_key if self.config.gcp_profile.kms_key else None
```
Comment (jsilva) [inline src/wanna/core/services/jobs.py:436]: we should load the model `encryption_spec_key_name` and then fallback to `encryption_spec_key_name`

File: src/wanna/core/deployment/models.py [hunk 2]
```diff
@@ -92,6 +92,7 @@ class JobResource(GCPResource, Generic[JOB]):
     tensorboard: Optional[str]
     network: str
     job_config: JOB
+    encryption_spec_key_name: Optional[str]
```
Comment (jsilva) [inline src/wanna/core/deployment/models.py:95]: I would expect this to be used somewhere in `vertex_jobs.py`

Comment (michalmrazek) [reply src/wanna/core/deployment/models.py:95]: I agree, it doesnt seem to be used anywhere?

**Pull request merged**